Issue with Jest dependencies (@nrwl/jest, @jest/types and @jest/reporters)

### Current Behavior

Currently unable to run `npm i` due to dependency issues between @nrwl/jest, @jest/types and @jest/reporters. 

ts-jest@29.0.5 wants to use @jest/types@"^29.0.0"  but @jest/reporters@28.1.1 is bringing in @jest/types@28.1.3. @nrwl/jest@15.9.1 depends on @jest/reporters@28.1.1.


### Expected Behavior

- npm i installs packages
- @nrwl/jest@15.9.1 updated to depend on later version of @jest/reporters

### GitHub Repo

_No response_

### Steps to Reproduce

1. run`npm i` with v15.9.1


### Nx Report

```shell
Node : 16.14.0
   OS   : darwin x64
   npm  : 7.10.0

   nx                      : 15.9.1
   @nrwl/js                : 15.9.1
   @nrwl/jest              : 15.9.1
   @nrwl/linter            : 15.9.1
   @nrwl/workspace         : 15.9.1
   @nrwl/angular           : 15.9.1
   @nrwl/cli               : 15.9.1
   @nrwl/cypress           : 15.9.1
   @nrwl/devkit            : 15.9.1
   @nrwl/eslint-plugin-nx  : 15.9.1
   @nrwl/react             : 15.9.1
   @nrwl/tao               : 15.9.1
   @nrwl/web               : 15.9.1
   @nrwl/webpack           : 15.9.1
   typescript              : 4.9.5
   ---------------------------------------
   Community plugins:
   @ngrx/component-store : 15.3.0
   @ngrx/effects         : 15.3.0
   @ngrx/entity          : 15.3.0
   @ngrx/router-store    : 15.3.0
   @ngrx/store           : 15.3.0
   rxjs                  : 6.6.7
   @compodoc/compodoc    : 1.1.19
   @ngrx/schematics      : 15.3.0
   @ngrx/store-devtools  : 15.3.0
```


### Failure Logs

```shell
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! Found: @jest/types@28.1.3
npm ERR! node_modules/@jest/types
npm ERR!   @jest/types@"^28.1.1" from @jest/reporters@28.1.1
npm ERR!   node_modules/@jest/reporters
npm ERR!     @jest/reporters@"28.1.1" from @nrwl/jest@15.9.1
npm ERR!     node_modules/@nrwl/jest
npm ERR!       dev @nrwl/jest@"15.9.1" from the root project
npm ERR!       1 more (@nrwl/angular)
npm ERR!   @jest/types@"^28.1.1" from @jest/test-result@28.1.1
npm ERR!   node_modules/@jest/test-result
npm ERR!     @jest/test-result@"28.1.1" from @nrwl/jest@15.9.1
npm ERR!     node_modules/@nrwl/jest
npm ERR!       dev @nrwl/jest@"15.9.1" from the root project
npm ERR!       1 more (@nrwl/angular)
npm ERR!     @jest/test-result@"^28.1.1" from @jest/reporters@28.1.1
npm ERR!     node_modules/@jest/reporters
npm ERR!       @jest/reporters@"28.1.1" from @nrwl/jest@15.9.1
npm ERR!       node_modules/@nrwl/jest
npm ERR!         dev @nrwl/jest@"15.9.1" from the root project
npm ERR!         1 more (@nrwl/angular)
npm ERR!   27 more (jest-config, jest-util, @jest/console, @jest/transform, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peerOptional @jest/types@"^29.0.0" from ts-jest@29.0.5
npm ERR! node_modules/ts-jest
npm ERR!   dev ts-jest@"29.0.5" from the root project
npm ERR!   ts-jest@"^29.0.0" from jest-preset-angular@13.0.0
npm ERR!   node_modules/jest-preset-angular
npm ERR!     dev jest-preset-angular@"13.0.0" from the root project
```


### Additional Information

_No response_

--- a/packages/jest/package.json
+++ b/packages/jest/package.json
@@ -34,17 +34,17 @@
     "migrations": "./migrations.json"
   },
   "dependencies": {
-    "@jest/reporters": "28.1.1",
-    "@jest/test-result": "28.1.1",
+    "@jest/reporters": "^29.4.1",
+    "@jest/test-result": "^29.4.1",
     "@nrwl/devkit": "file:../devkit",
     "@nrwl/js": "file:../js",
     "@phenomnomnominal/tsquery": "~5.0.1",
     "chalk": "^4.1.0",
     "dotenv": "~10.0.0",
     "identity-obj-proxy": "3.0.0",
-    "jest-config": "28.1.1",
-    "jest-resolve": "28.1.1",
-    "jest-util": "28.1.1",
+    "jest-config": "^29.4.1",
+    "jest-resolve": "^29.4.1",
+    "jest-util": "^29.4.1",
     "resolve.exports": "1.1.0",
     "tslib": "^2.3.0"
   },

